dawn/src/dawn/Validator/UnstructuredDimensionChecker.h

```diff
@@ -28,11 +29,22 @@ class UnstructuredDimensionChecker {
 
 private:
   class UnstructuredDimensionCheckerImpl : public ast::ASTVisitorForwarding {
+  public:
+    struct UnstructuredDimensionCheckerConfig {
```

Comment:
why is this declaration public?

Reply:
maybe I'm missing something obvious here, but the `UnstructuredDimensionCheckerConfig` object is passed to the constructor of the `UnstructuredDimensionCheckerImpl`. So a caller wishing to configure the `UnstructuredDimensionCheckerImpl` has to have access to the `UnstructuredDimensionCheckerConfig`. Would you prefer to have this private and make `UnstructuredDimensionChecker` a friend class?

Reply:
got it, I was confused. This is fine